jupyter-extension/jupyterlab_pachyderm/handlers.py

```diff
@@ -648,16 +649,36 @@ def write_config(
 
 
 def setup_handlers(web_app, config_file: Path):
+    """
+    Sets up the Pachyderm client and the HTTP request handler.
+
+    Config for the Pachyderm client will first be attempted by reading
+    the local config file. This falls back to the PACHD_ADDRESS and
+    DEX_TOKEN env vars, and finally defaulting to a localhost client
+    on the default port 30650 failing that.
+    """
+    client = None
     try:
         client = Client().from_config(config_file)
         get_logger().debug(
             f"Created Pachyderm client for {client.address} from local config"
         )
     except FileNotFoundError:
-        get_logger().debug(
-            "Could not find config file -- no pachyderm client instantiated"
-        )
-    else:
+        if PACHD_ADDRESS:
+            client = Client().from_pachd_address(pachd_address=PACHD_ADDRESS)
+            if DEX_TOKEN:
+                client.auth_token = client.auth.authenticate(
+                    id_token=DEX_TOKEN
+                ).pach_token
+            get_logger().debug(
+                f"Created Pachyderm client for {client.address} from env var"
+            )
+        else:
+            get_logger().debug(
+                "Could not find config file -- no pachyderm client instantiated"
+            )
```

Comment:
Do we want to add any tests for this new fallback behavior? I think we avoided testing setup_handlers before since it was so simple, but maybe given these branching paths it could be worthwhile now?

Reply:
For the sake of getting this into the release tomorrow, feel free to handle this in a follow up if you end up thinking it is necessary.

Reply:
i will file a ticket for this, i don't think it's very straightforward to toggle the env vars in our testing set up right now.